src/common/task.cc

```diff
@@ -375,14 +375,31 @@ TaskExecutionSpec::TaskExecutionSpec(
     int64_t task_spec_size) {
   execution_dependencies_ = execution_dependencies;
   task_spec_size_ = task_spec_size;
+  spillback_count_ = 0;
+  last_tstamp_ = current_time_ms();
```

Comment:
How expensive is this? I wonder if this is too much overhead.

Reply:
yeah, I'll time it. I'm leaning towards setting the timestamp to zero in the constructor and then explicitly setting it only when needed (i.e., at the two local scheduler entry points).